From 10c62f784e040e9f7f07f9a3e41c3777a04d91e5 Mon Sep 17 00:00:00 2001
From: Dennis Rassmann <showp1984@gmail.com>
Date: Mon, 11 Nov 2013 02:08:06 +0100
Subject: [PATCH 250/338] net/bridge/br_multicast.c: fix:
 include/net/ipv6.h:362:27: warning: '*((void
 *)&ip+4)'& '*((void *)&ip+8)' may be used
 uninitialized in this function

Signed-off-by: Dennis Rassmann <showp1984@gmail.com>
Signed-off-by: Simarpreet Singh <simar@linux.com>
---
 net/bridge/br_multicast.c |    6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/net/bridge/br_multicast.c b/net/bridge/br_multicast.c
index 27ca25e..8044aff 100644
--- a/net/bridge/br_multicast.c
+++ b/net/bridge/br_multicast.c
@@ -139,6 +139,12 @@ struct net_bridge_mdb_entry *br_mdb_get(struct net_bridge *br,
 {
 	struct net_bridge_mdb_htable *mdb = rcu_dereference(br->mdb);
 	struct br_ip ip;
+	int i = 0;
+	ip.u.ip4 = 0;
+
+	for (i=0; i<16; i++)
+		ip.u.ip6.s6_addr[i] = 0;
+
 
 	if (br->multicast_disabled)
 		return NULL;
-- 
1.7.9.5

